analyzers/src/SonarAnalyzer.Common/Rules/NonStandardCryptographicAlgorithmsShouldNotBeUsedBase.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class NonStandardCryptographicAlgorithmsShouldNotBeUsedBase<TSyntaxKind, TTypeDeclarationSyntax> : HotspotDiagnosticAnalyzer
+        where TTypeDeclarationSyntax : SyntaxNode
+        where TSyntaxKind : struct
+    {
+        internal const string DiagnosticId = "S2257";
+
+        protected const string MessageFormat = "Make sure using a non-standard cryptographic algorithm is safe here.";
+
+        private readonly ImmutableArray<KnownType> nonInheritableClassesAndInterfaces = ImmutableArray.Create<KnownType>(
+            new[]
```

Comment:
pedantic: There is no need to create an array for the types, we can pass them directly as parameters.